R/prepare_matches_mapper.R

```diff
@@ -13,9 +13,10 @@ prepare_matches_mapper <- function(mapper, activities) {
     left_join(activities, by = "activity_uuid_product_uuid") |>
     summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered")) |>
     ungroup() |>
+    mutate(across(everything(), ~ifelse(. == "", NA_character_, .))) |>
     distinct() |>
     mutate(multi_match = case_when(
-      multi_match == "" ~ "FALSE",
+      is.na(multi_match) ~ "FALSE",
```

Comment:
So we expect the character "FALSE" not the logical `FALSE`? See the test below.

Reply:
@maurolepore this line of code:
```summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered"))```
converts the `multi_match` column type from logical to character. However, there is no need to keep the column type to logical. Hence, I am choosing to convert it into character type. Please let me know if you think otherwise! :)

Reply:
``` r
# Good
isTRUE(TRUE)
#> [1] TRUE
testthat::expect_true(TRUE)

# Confusing
isTRUE("TRUE")
#> [1] FALSE
testthat::expect_true("TRUE")
#> Error: "TRUE" is not TRUE
#> 
#> `actual` is a character vector ('TRUE')
#> `expected` is a logical vector (TRUE)
```

<sup>Created on 2023-10-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
``` r
# Logicals allow you to do useful things
good <- tibble::tibble(x = c(TRUE, TRUE, FALSE))

good |> dplyr::filter(x)
#> # A tibble: 2 × 1
#>   x    
#>   <lgl>
#> 1 TRUE 
#> 2 TRUE

good |> dplyr::filter(!x)
#> # A tibble: 1 × 1
#>   x    
#>   <lgl>
#> 1 FALSE

sum(good$x)
#> [1] 2

mean(good$x)
#> [1] 0.6666667



# Characters are very differnt

bad <- tibble::tibble(x = c("TRUE", "TRUE", "FALSE"))

try(bad |> dplyr::filter(x))
#> Error in dplyr::filter(bad, x) : ℹ In argument: `x`.
#> Caused by error:
#> ! `..1` must be a logical vector, not a character vector.

try(bad |> dplyr::filter(!x))
#> Error in dplyr::filter(bad, !x) : ℹ In argument: `!x`.
#> Caused by error in `!x`:
#> ! invalid argument type

try(sum(bad$x))
#> Error in sum(bad$x) : invalid 'type' (character) of argument

try(mean(bad$x))
#> Warning in mean.default(bad$x): argument is not numeric or logical: returning
#> NA
#> [1] NA
```

<sup>Created on 2023-10-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
@maurolepore I now see a good reason to keep the column type to logical. Thanks for explaining its importance! Please review the new change! Thanks!